test/gatekeeper_controller_test.go

```diff
@@ -62,33 +127,314 @@ var _ = Describe("Gatekeeper", func() {
+				Expect(webhookDeployment.Spec.Replicas).NotTo(BeNil())
+				Expect(webhookDeployment.Spec.Replicas).To(Equal(gatekeeper.Spec.Webhook.Replicas))
+			})
+
+			By("Checking expected pod affinity", func() {
+				Expect(auditDeployment.Spec.Template.Spec.Affinity).To(BeEquivalentTo(gatekeeper.Spec.Affinity))
+				Expect(webhookDeployment.Spec.Template.Spec.Affinity).To(BeEquivalentTo(gatekeeper.Spec.Affinity))
+			})
+
+			By("Checking expected node selector", func() {
+				Expect(auditDeployment.Spec.Template.Spec.NodeSelector).To(BeEquivalentTo(gatekeeper.Spec.NodeSelector))
+				Expect(webhookDeployment.Spec.Template.Spec.NodeSelector).To(BeEquivalentTo(gatekeeper.Spec.NodeSelector))
+			})
+
+			By("Checking expected pod annotations", func() {
+				Expect(auditDeployment.Spec.Template.Annotations).To(BeEquivalentTo(gatekeeper.Spec.PodAnnotations))
+				Expect(webhookDeployment.Spec.Template.Annotations).To(BeEquivalentTo(gatekeeper.Spec.PodAnnotations))
+			})
+
+			By("Checking expected tolerations", func() {
+				Expect(auditDeployment.Spec.Template.Spec.Tolerations).To(BeEquivalentTo(gatekeeper.Spec.Tolerations))
+				Expect(webhookDeployment.Spec.Template.Spec.Tolerations).To(BeEquivalentTo(gatekeeper.Spec.Tolerations))
+			})
+
+			By("Checking expected resource limits and requests", func() {
+				assertResources(*gatekeeper.Spec.Resources, auditDeployment.Spec.Template.Spec.Containers[0].Resources)
+				assertResources(*gatekeeper.Spec.Resources, webhookDeployment.Spec.Template.Spec.Containers[0].Resources)
+			})
+
+			By("Checking expected image", func() {
+				Expect(auditDeployment.Spec.Template.Spec.Containers[0].Image).To(Equal(*gatekeeper.Spec.Image.Image))
+				Expect(auditDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy).To(Equal(*gatekeeper.Spec.Image.ImagePullPolicy))
+				Expect(webhookDeployment.Spec.Template.Spec.Containers[0].Image).To(Equal(*gatekeeper.Spec.Image.Image))
+				Expect(webhookDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy).To(Equal(*gatekeeper.Spec.Image.ImagePullPolicy))
+			})
+
+			By("Checking expected failure policy", func() {
+				validatingWebhookConfiguration := &admregv1.ValidatingWebhookConfiguration{}
+				Eventually(func() error {
+					return K8sClient.Get(ctx, validatingWebhookName, validatingWebhookConfiguration)
+				}, waitTimeout, pollInterval).ShouldNot(HaveOccurred())
+				Expect(validatingWebhookConfiguration).NotTo(BeNil())
+				for _, wh := range validatingWebhookConfiguration.Webhooks {
+					if wh.Name == "validation.gatekeeper.sh" {
+						Expect(wh.FailurePolicy).To(Equal(gatekeeper.Spec.Webhook.FailurePolicy))
+					}
+				}
+			})
+
+			By("Checking expected audit interval", func() {
+				value, found := getContainerArg(auditDeployment.Spec.Template.Spec.Containers[0].Args, "--audit-interval")
+				Expect(found).To(BeTrue())
+				Expect(value).To(Equal("--audit-interval=10"))
```

Comment:
What do you think if we use the values from `gatekeeper_with_all_values.yaml`?

Reply:
Similar to unit tests, I think it's better to assert against the expected literal value instead of do type conversions and string concatenations.